src/bin/ipa_bench/config.rs

```diff
@@ -0,0 +1,127 @@
+use log::error;
+use rand_distr::num_traits::ToPrimitive;
+use serde_json::Value;
+use std::io::{BufReader, Read};
+use std::ops::Range;
+use std::process;
+
+#[derive(Clone)]
```

Comment:
Why isn't this directly serializable?  I know that you would need to use named types rather than tuples, but that's probably a good thing, isn't it?

It seems like you get a lot of custom serialization code here, where this could be annotations on the types instead.

The only caveat is for bounds checking, which I don't see in your code.  Most of the f64 values should be positive (not zero or negative), but that isn't checked.  CVR should be a value between 0 and 1, but you don't check that.

It's OK if a tool like this crashes when being provided bad input, but it's something to note in documentation.